File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnLookupLocalStreamIngest.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

public class GnLookupLocalStreamIngest : GnObject {
  private HandleRef swigCPtr;

  internal GnLookupLocalStreamIngest(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnLookupLocalStreamIngest_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnLookupLocalStreamIngest obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnLookupLocalStreamIngest() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnLookupLocalStreamIngest(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnLookupLocalStreamIngest(GnLookupLocalStreamIngestEventsDelegate pEventDelegate) : this(gnsdk_csharp_marshalPINVOKE.new_GnLookupLocalStreamIngest(GnLookupLocalStreamIngestEventsDelegate.getCPtr(pEventDelegate)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
* Write to the ingestion process . This can be called multiple times to ensure that data is written as and when it is available.
*  @param data             [in] data to ingest
*  @param dataLength       [in] size of data being written
*/
  public void Write(byte[] bundleData, uint dataSize) {
    gnsdk_csharp_marshalPINVOKE.GnLookupLocalStreamIngest_Write(swigCPtr, bundleData, dataSize);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
* Flushes the memory cache to the file storage and commits the changes. This call will result in IO .
* Use this method to ensure that everything written is commited to the file system.
* Note: This is an optional call as internally data is flushed when it exceed the cache size and when the object goes out of scope..
*/
  public void Flush() {
    gnsdk_csharp_marshalPINVOKE.GnLookupLocalStreamIngest_Flush(swigCPtr);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnLookupLocalStreamIngestEventsDelegate EventHandler() {
    IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnLookupLocalStreamIngest_EventHandler(swigCPtr);
    GnLookupLocalStreamIngestEventsDelegate ret = (cPtr == IntPtr.Zero) ? null : new GnLookupLocalStreamIngestEventsDelegate(cPtr, false);
    return ret;
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnLookupLocalStreamIngest.java

package com.gracenote.gnsdk;

public class GnLookupLocalStreamIngest extends GnObject {
  private long swigCPtr;

  protected GnLookupLocalStreamIngest(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnLookupLocalStreamIngest_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnLookupLocalStreamIngest obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnLookupLocalStreamIngest(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  private IGnLookupLocalStreamIngestEvents 			pEventDelegate;
  private IGnLookupLocalStreamIngestEventsProxyU 	eventHandlerProxy;

/** 
*  Constructor for creating a {@link GnLookupLocalStreamIngest} Object 
*  @param pEventDelegate           [in] Delegate to receive events . 
*  @param callbackData             [in] Optional user data that will be sent to the Delegate 
*/ 
 
  public GnLookupLocalStreamIngest(IGnLookupLocalStreamIngestEvents pEventDelegate) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnLookupLocalStreamIngestEventsProxyU(pEventDelegate);
	this.pEventDelegate = pEventDelegate; // <REFERENCE_NAME_CHECK><TYPE>IGnLookupLocalStreamIngestEvents</TYPE><NAME>pEventDelegate</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
	swigCPtr = gnsdk_javaJNI.new_GnLookupLocalStreamIngest((eventHandlerProxy==null)?0:IGnLookupLocalStreamIngestEventsProxyL.getCPtr(eventHandlerProxy), eventHandlerProxy);
}

  public void write(byte[] bundleData, long dataSize) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnLookupLocalStreamIngest_write(swigCPtr, this, bundleData, dataSize);
  }

/** 
* Flushes the memory cache to the file storage and commits the changes. This call will result in IO . 
* Use this method to ensure that everything written is commited to the file system. 
* Note: This is an optional call as internally data is flushed when it exceed the cache size and when the object goes out of scope.. 
*/ 
 
  public void flush() throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnLookupLocalStreamIngest_flush(swigCPtr, this);
  }

  public IGnLookupLocalStreamIngestEvents eventHandler() {
	return pEventDelegate;
}

}
